statemachine/src/test/java/com/amazonaws/stepfunctions/cloudformation/statemachine/DeleteHandlerTest.java

```diff
@@ -51,76 +99,68 @@ public void testSuccess() {
 
     @Test
     public void test400() {
-        Mockito.when(proxy.injectCredentialsAndInvoke(Mockito.any(DeleteStateMachineRequest.class), Mockito.any(Function.class))).thenThrow(exception400);
+        Mockito.when(proxy.injectCredentialsAndInvoke(Mockito.any(), Mockito.any(Function.class))).thenThrow(exception400);
```

Comment:
why get less specific with the type? this loosens the test criteria... is that really necessary given that it's in the `DeleteHandlerTest`?

Reply:
Now that the delete handler calls both DescribeStateMachine then DeleteStateMachine, these tests were failing. I've updated them to first return a successful response for DescribeStateMachine, and then a 400 or 500 for DeleteStateMachine, which is the behaviour which was previously being tested.